tests/integ/test_df_to_snowpark_pandas.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+# Tests behavior of to_snowpark_pandas() without explicitly initializing Snowpark pandas.
+
+import pytest
+
+from snowflake.snowpark._internal.utils import TempObjectType
+from tests.utils import Utils
+
+
+@pytest.fixture(scope="module")
+def tmp_table_basic(session):
+    table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+    Utils.create_table(
+        session, table_name, "id integer, foot_size float, shoe_model varchar"
+    )
+    session.sql(f"insert into {table_name} values (1, 32.0, 'medium')").collect()
+    session.sql(f"insert into {table_name} values (2, 27.0, 'small')").collect()
+    session.sql(f"insert into {table_name} values (3, 40.0, 'large')").collect()
+
+    try:
+        yield table_name
+    finally:
+        Utils.drop_table(session, table_name)
+
+
+def test_to_snowpark_pandas_no_modin(session, tmp_table_basic):
+    snowpark_df = session.table(tmp_table_basic)
+    # Check if modin is installed
+    try:
+        import modin  # noqa: F401
```

Comment:
why do we import modin here ?

Reply:
I misunderstood how our test environments are set up and I thought the `modin-*` CI jobs would also run Snowpark Python tests, so I added this to check if modin was present in the current environment. It seems like this is currently never run with modin installed, so we'd have to explicitly create separate environments to run this test (see my comment in the other thread).

Reply:
I changed the test setup such that this test runs in both Snowpark Python and Snowpark pandas CI. The `import modin` statement is used to check which environment we're in; apparently this try/except is the "idiomatic" way of checking for the presence of a package in Python.